src/libxtp/gwbse/rpa.cc

```diff
@@ -113,11 +115,17 @@ RPA::rpa_eigensolution RPA::Diagonalize_H2p() const {
   sol.omega = Eigen::VectorXd::Zero(es.eigenvalues().size());
   sol.omega = es.eigenvalues().cwiseSqrt();
 
+  RPA_correlation_energy += 0.5 * sol.omega.sum();
+
   XTP_LOG(Log::info, _log) << TimeStamp()
                            << " Lowest neutral excitation energy (eV): "
                            << tools::conv::hrt2ev * sol.omega.minCoeff()
                            << std::flush;
 
+  XTP_LOG(Log::error, _log) << TimeStamp()
+                           << " RPA correlation energy (Hartree): "
+                           << RPA_correlation_energy << std::flush;
```

Comment:
could you add a comment  where this is coming from?

Reply:
How the energy is calculated? Eq.9 same paper -1/2 Tr(HRPA^1/2 - A) as a code comment or output info ?

Reply:
comment in the code with the paper is good.

Reply:
would be also nice to have some sort of test for this.

Reply:
Not sure how since this is not stored/reported anywhere outside this method

Reply:
Unless we add an energy variable to `struct rpa_eigensolution`

Reply:
test added